Bitkit/ViewModels/Extensions/SettingsViewModel+Electrum.swift

```diff
@@ -0,0 +1,179 @@
+import Foundation
+
+// MARK: - Electrum Server Management
+
+extension SettingsViewModel {
+    func loadElectrumSettings() {
+        // Check connection status first
+        let isRunning = lightningService.status?.isRunning == true
+        electrumIsConnected = isRunning
+
+        // Update form with current server (stored or default)
+        let currentServer = electrumConfigService.getCurrentServer()
+        updateForm(with: currentServer)
+    }
+
+    func resetElectrumToDefault() async -> (success: Bool, host: String, port: String, errorMessage: String?) {
+        let defaultServer = electrumConfigService.getDefaultServer()
+        updateForm(with: defaultServer)
+
+        return await connectToElectrumServer()
+    }
+
+    func connectToElectrumServer() async -> (success: Bool, host: String, port: String, errorMessage: String?) {
+        electrumIsLoading = true
+
+        let host = electrumHost.trimmingCharacters(in: .whitespaces)
+        let port = electrumPort.trimmingCharacters(in: .whitespaces)
+
+        // Validate input first
+        if let validationError = validateElectrumInput(host: host, port: port) {
+            electrumIsLoading = false
+            return (success: false, host: host, port: port, errorMessage: validationError)
+        }
+
+        // Update current server config immediately after validation
+        electrumCurrentServer = ElectrumServer(
+            host: host,
+            portString: port,
+            protocolType: electrumSelectedProtocol
+        )
+
+        do {
+            // Save the configuration to settings
+            electrumConfigService.saveServerConfig(electrumCurrentServer)
+
+            // Restart the Lightning node with the new Electrum server
+            try await lightningService.restartWithElectrumServer(electrumCurrentServer.url)
+
+            electrumIsConnected = true
+            electrumIsLoading = false
+
+            Logger.info("Successfully connected to Electrum server: \(electrumCurrentServer.url)")
+
+            return (success: true, host: host, port: port, errorMessage: nil)
+        } catch {
+            electrumIsConnected = false
+            electrumIsLoading = false
+
+            Logger.error(error, context: "Failed to connect to Electrum server")
+
+            return (success: false, host: host, port: port, errorMessage: nil)
```

Comment:
In android we're also reconnecting to the previous connection if the new one failed.
Not sure if it's a must -- maybe that's the main use case of the "Reset to Default" button -- but perhaps an improvement idea for the future ;)

Reply:
Good idea but I'd rather leave the UX like it is in the current production app. I think this could easily be overlooked as the connection working when it in fact failed at fell back to default.